app/javascript/map.js

```diff
@@ -52,6 +54,8 @@ class Map {
     });
     mapOptions.center = new google.maps.LatLng(center[0], center[1]);
 
+    bounds = new google.maps.LatLngBounds();
```

Comment:
Why don't you get these bounds from the map?  Right now these are totally independent from the map's center.